File: go/day55.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
	"regexp"
)

func main() {
	fmt.Println("Please input your phone number")

	reader := bufio.NewReader(os.Stdin)

	phone,_ := reader.ReadString('\n')
	phone = strings.TrimSpace(phone)

	match, _ := regexp.MatchString("^[0-9]{8}$", phone)

	fmt.Println("Valid phone number: ", match)
}

File: cSharp/day55.cs
using System;
using System.Text.RegularExpressions;

Console.WriteLine("Please input your phone number");
string password = Console.ReadLine();

Regex hasCorrectNumberOfDigits = new Regex("^\\d{8}$");

if (
    hasCorrectNumberOfDigits.Match(password).Success
) {
    Console.WriteLine("phone number is valid");
}
else {
    Console.WriteLine("phone number is INVALID");
}